dask_sql/input_utils/location.py

```diff
@@ -26,17 +31,20 @@ def to_dc(
 
         if format == "memory":
             client = default_client()
-            return client.get_dataset(input_item, **kwargs)
+            df = client.get_dataset(input_item, **kwargs)
+
+            if gpu and dask_cudf and not isinstance(df, dask_cudf.DataFrame):
+                return dask_cudf.from_dask_dataframe(df)
```

Comment:
I think it's a reasonable expectation that if we have a published dataframe on host and we create a new table from it with `gpu = True`, a dataframe on GPU should be the result. Should this apply in the opposite scenario? Essentially what should be returned here:

```python
    client.publish_dataset(df=dask_cudf.from_cudf(df, npartitions=1))

    c.sql(
        f"""
        CREATE TABLE
            new_table
        WITH (
            location = 'df',
            format = 'memory',
            gpu = False
        )
    """
    )

    return_df = c.sql(
        """
        SELECT * FROM new_table
    """
    ).compute()
```

Reply:
This brings up a good point. Looking at this [branch](https://github.com/rapidsai/dask-sql/blob/branch-21.12/dask_sql/input_utils/pandaslike.py#L41) in the pandas like module it looks like when we set gpu=False we still return a dask_cudf dataframe.

So in this case I guess we can continue with that behavior. Ideally we'd like to warn users when this is happening to avoid confusion, but can be done in a followup PR for both this as well as `pandaslike`